knime_extension/src/nodes/locationanalysis.py

```diff
@@ -633,3 +681,548 @@ def getIndex(s):
+        for j in range(candidate):
+            dfy.loc[j, _Chosen] = pulp.value(y[j])
+        dfy[_FACILITYID] = dfy[_FACILITYID].astype(str)
+        dfy[_Chosen] = dfy[_Chosen].astype(int)
+        return knext.Table.from_pandas(dfy)
+
+
+############################################
+# Location-allocation LSCP Solver
+############################################
+@knext.node(
+    name="LSCP Solver",
+    node_type=knext.NodeType.MANIPULATOR,
+    icon_path=__NODE_ICON_PATH + "LSCP.png",
+    category=__category,
+    after="",
+)
+@knext.input_table(
+    name="Input demand data",
+    description="Table including a demand column and an optional distance column to the closest required facilities.",
+)
+@knext.input_table(
+    name="Input OD Matrix for candidate facilities",
+    description="Table consisting of columns (totaling n+1 columns) for demand ID and the distance matrix to the candidate facilities (n).",
+)
+@knext.output_table(
+    name="LSCP result table",
+    description="candidate column names and chosen status(1/0)",
+)
+@knut.pulp_node_description(
+    short_description="Solve LSCP problem to minimize the number of facilities within an impedance cutoff.",
+    description="""The LSCP model,location set covering problem [(LSCP)] (https://en.wikipedia.org/wiki/Facility_location_problem), 
+     aims to cover all demand points within a threshold distance. 
+
+    The node requires two inputs. The first (top) input should contain the column(s): demand ID and an optional column for distance to closest required facilities. 
+    The second (bottom) input should only include the demand ID column which is identical to the one from the top input, 
+    along with columns representing the distance matrix to candidate facilities (n).It has a dimension of m x (n+1).
+
+    
+    The LSCP problem will be solved by PuLP package. 
+    """,
+    references={
+        "Pulp.Solver": "https://coin-or.github.io/pulp/guides/how_to_configure_solvers.html",
+    },
+)
+class LSCPSolverNode:
+    id_left = knext.ColumnParameter(
+        "Demand ID column from demand table",
+        "The column indicating the demand location IDs.",
+        # port_index=0,
+        column_filter=knut.is_int_or_string,
+        include_row_key=False,
+        include_none_column=False,
+    )
+
+    requiredid = knext.ColumnParameter(
+        "Distance column for required facilities",
+        "The travel cost to the nearest required facilities ",
+        # port_index=0,
+        column_filter=knut.is_numeric,
+        include_row_key=False,
+        include_none_column=True,
+    )
+    id_right = knext.ColumnParameter(
+        "Demand ID column from distance matrix table",
+        "The column indicating the demand location IDs.",
+        port_index=1,
+        column_filter=knut.is_int_or_string,
+        include_row_key=False,
+        include_none_column=False,
+    )
+    threshold = knext.DoubleParameter(
+        "Threshold",
```

Comment:
Add validation

Reply:
Can we revise the validation function for p for this one? I add the validating of P